spec/integration/indirector/file_metadata/file_server_spec.rb

```diff
@@ -4,11 +4,147 @@
 require 'puppet/indirector/file_metadata/file_server'
 require 'shared_behaviours/file_server_terminus'
 
+require 'puppet_spec/files'
+
 describe Puppet::Indirector::FileMetadata::FileServer, " when finding files" do
   it_should_behave_like "Puppet::Indirector::FileServerTerminus"
+  include PuppetSpec::Files
+  CHECKSUMS = [['md5', '{md5}a5ea0ad9260b1550a14cc58d2c39b03d'],
+               ['md5lite', '{md5lite}a5ea0ad9260b1550a14cc58d2c39b03d'],
+               ['sha256', '{sha256}f1b2f662800122bed0ff255693df89c4487fbdcf453d3524a42d4ec20c3d9c04'],
+               ['sha256lite', '{sha256lite}f1b2f662800122bed0ff255693df89c4487fbdcf453d3524a42d4ec20c3d9c04']]
 
   before do
     @terminus = Puppet::Indirector::FileMetadata::FileServer.new
     @test_class = Puppet::FileServing::Metadata
+    Puppet::FileServing::Configuration.instance_variable_set(:@configuration, nil)
+  end
+
+  describe "with a plugin environment specified in the request" do
+    before do
+      path = tmpfile("file_content_with_env")
+
+      Dir.mkdir(path)
+
+      modpath = File.join(path, "mod")
+      FileUtils.mkdir_p(File.join(modpath, "lib"))
+      file = File.join(modpath, "lib", "file.rb")
+      File.open(file, "wb") { |f| f.write "1\r\n" }
```

Comment:
Yup, going to add a test for longer file lengths.

On Fri, Jan 23, 2015 at 11:24 AM, Josh Cooper <notifications@github.com>
wrote:

> In spec/integration/indirector/file_metadata/file_server_spec.rb
> <https://github.com/puppetlabs/puppet/pull/3507#discussion_r23472850>:
>
> >
> >    before do
> >      @terminus = Puppet::Indirector::FileMetadata::FileServer.new
> >      @test_class = Puppet::FileServing::Metadata
> > +    Puppet::FileServing::Configuration.instance_variable_set(:@configuration, nil)
> > +  end
> > +
> > +  describe "with a plugin environment specified in the request" do
> > +    before do
> > +      path = tmpfile("file_content_with_env")
> > +
> > +      Dir.mkdir(path)
> > +
> > +      modpath = File.join(path, "mod")
> > +      FileUtils.mkdir_p(File.join(modpath, "lib"))
> > +      file = File.join(modpath, "lib", "file.rb")
> > +      File.open(file, "wb") { |f| f.write "1\r\n" }
>
> [image: :+1:] to including "\r" to test binary mode.
>
> I'm thinking we should increase the length of this file, so that we can
> test that the "lite" checksums are different than their non-lite
> counterparts.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/3507/files#r23472850>.
>